lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -689,23 +635,21 @@ export default class BitcoinProcessor {
     for (let i = 0 ; i < transaction.inputs.length ; i++) {
 
       const currentInput = transaction.inputs[i];
-      const transactionIdForCurrentInput = currentInput.prevTxId.toString('hex');
-
-      const transactionOutValue = await this.getTransactionOutValueInSatoshi(transactionIdForCurrentInput, currentInput.outputIndex);
+      const transactionOutValue = await this.getTransactionOutValueInSatoshi(currentInput.previousTransactionId, currentInput.outputIndexInPreviousTransaction);
 
       inputSatoshiSum += transactionOutValue;
     }
 
     // transaction outputs in satoshis
-    const transactionOutputs: number[] = transaction.outputs.map((output) => Math.round(output.satoshis));
+    const transactionOutputs: number[] = transaction.outputs.map((output) => output.satoshis);
 
     const outputSatoshiSum = transactionOutputs.reduce((sum, value) => sum + value, 0);
 
     return (inputSatoshiSum - outputSatoshiSum);
   }
 
   private async addValidSidetreeTransactionsFromVOutsToTransactionStore (
```

Comment:
Design: what if we change this method to be `getSidetreeTransactions(blockData)`, then:

1. Method becomes much more intentional.
1.  Loop logic within in `processBlock` from line 592 to 614 becomes encapsulated in this method, making `processBlock` itself much more concise thus easier to understand (Adding to MongoDB will be done in `processBlock`).

Reply:
Agree with you 100%. I didn't want to do this change as I thought that this change will make this changelist bigger but looks like that you understand the context here so I can make this change now.